php/media/class-gallery.php

```diff
@@ -398,6 +396,39 @@ protected function register_settings() {
 		$media_settings->add_setting( $this->settings );
 	}
 
+	/**
+	 * Inits the cloudinary gallery using block attributes.
+	 *
+	 * @param string $content The post content.
+	 * @param array  $block   Block data.
+	 *
+	 * @return string
+	 */
+	public function prepare_block_render( $content, $block ) {
+		if ( 'cloudinary/gallery' !== $block['blockName'] ) {
+			return $content;
+		}
+
+		$attributes                = Utils::expand_dot_notation( $block['attrs'], '_' );
+		$attributes                = array_merge( self::$default_config, $attributes );
+		$attributes['mediaAssets'] = $attributes['selectedImages'];
+		$attributes['cloudName']   = $this->media->plugin->components['connect']->get_cloud_name();
+		unset( $attributes['selectedImages'], $attributes['customSettings'] );
+
+		ob_start();
+		?>
+<script>
+	(function () {
+		const attributes = JSON.parse( '<?php echo wp_json_encode( $attributes ); ?>' );
+		attributes.container = '.' + attributes.container;
+		cloudinary.galleryWidget( attributes ).render();
```

Comment:
This kind of direct function call requires that the JS API is already loaded and defined somewhere above in the HTML DOM. Could we use a queue system instead where all of the widget config gets pushed to an array which is then parsed and loaded as widgets in the footer after the external API is loaded?

Reply:
That would require hooking into a high part of the DOM (ie. `wp_head`) in order to initialize an empty variable. I found a workaround for this by simply listening for the `load` event.